scripts/validate-codex-project-boundary.py

```
@@ -0,0 +1,198 @@
+
+def _read_response(process: subprocess.Popen[str], request_id: int) -> dict[str, Any]:
+    if process.stdout is None:
+        raise BoundaryError("Codex configuration probe stdout is unavailable")
+    deadline = time.monotonic() + STARTUP_TIMEOUT_SECONDS
+    while True:
+        remaining = deadline - time.monotonic()
+        if remaining <= 0:
+            raise BoundaryError("Codex configuration probe timed out")
+        ready, _, _ = select.select([process.stdout], [], [], remaining)
+        if not ready:
+            raise BoundaryError("Codex configuration probe timed out")
```

Ruff (0.16.3)

[warning] Avoid specifying long messages outside the exception class

(TRY003)